<launch>
  <arg name="port" default="/dev/ttyUSB0" />

  <!-- When you integrate into your platform, use the provided xacro macro to include
       the D46 URDF with your larger platform description, and have your platform's
       joint_state_publisher subscribe to the ptu state. -->
  <param name="robot_description" textfile="$(find flir_ptu_description)/urdf/example.urdf" /> 
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

  <node name="ptu_driver" pkg="flir_ptu_driver" type="ptu_node" ns="ptu" output="screen">
    <param name="connection_type" type="str" value="tcp" /> <!-- either tcp (uses ip_addr and tcp_port) or tty (uses tty port) -->
    <param name="ip_addr" type="str" value="128.18.40.195" /> 
    <param name="tcp_port" type="int" value="4000" /> 
    <param name="port" value="$(arg port)" /> <!-- tty port, like "/dev/ttyUSB0" -->
    <param name="test_pan_tilt_mode" type="bool" value="false"/> <!-- test mode sends random pan/tilt every few secs -->
    <!-- param name="ptu_control_mode" type="str" value="i" / --> <!-- PTU Speed control mode, either "i" or "v" (independent=default or pure velocity) -->
    <remap from="state" to="/joint_states" />
  </node>
</launch>
